File: core/utils/timer.go
package utils

import (
	"fmt"
	"math"
	"time"
)

func ConvertSecondsToHumanTime(seconds int) string {
	hours := math.Floor(float64(seconds) / 3600)
	minutes := math.Floor(float64(seconds%3600) / 60)
	remainingSeconds := seconds % 60

	formedHours := ""

	if hours < 10 {
		formedHours = fmt.Sprintf("0%d", int(hours))
	} else {
		formedHours = fmt.Sprintf("%d", int(hours))
	}

	formedMinutes := ""

	if minutes < 10 {
		formedMinutes = fmt.Sprintf("0%d", int(minutes))
	} else {
		formedMinutes = fmt.Sprintf("%d", int(minutes))
	}

	formedSeconds := ""

	if remainingSeconds < 10 {
		formedSeconds = fmt.Sprintf("0%d", int(remainingSeconds))
	} else {
		formedSeconds = fmt.Sprintf("%d", int(remainingSeconds))
	}

	return fmt.Sprintf("%s:%s:%s", formedHours, formedMinutes, formedSeconds)
}

func Bod(t time.Time) time.Time {
	year, month, day := t.Date()
	return time.Date(year, month, day, 0, 0, 0, 0, t.Location())
}

File: frontend/src/libraries/timer.ts
export function convertSecondsToHumanTime(seconds: number): string {
  const hours = Math.floor(seconds / 3600).toString().padStart(2, '0');
  const minutes = Math.floor((seconds % 3600) / 60).toString().padStart(2, '0');
  const remainingSeconds = (seconds % 60).toString().padStart(2, '0');

  return `${hours}:${minutes}:${remainingSeconds}`;
}

export function convertDateToInternationalString(date: Date): string {
  console.log(date);
  const year = date.getFullYear();
  const month = date.getMonth().toString().padStart(2, '0');
  const day = date.getDate().toString().padStart(2, '0');

  const hours = date.getHours().toString().padStart(2, '0');
  const minutes = date.getMinutes().toString().padStart(2, '0');
  const seconds = date.getSeconds().toString().padStart(2, '0');

  return `${year}/${month}/${day} ${hours}:${minutes}:${seconds}`; 
}
